src/components/Pagination/Pagination.tsx

```diff
@@ -1,45 +1,199 @@
-import { Button, ButtonGroup } from '@chakra-ui/react'
-import { ReactElement, useMemo } from 'react'
-import { generatePath, Link as RouterLink, useParams } from 'react-router-dom'
+import { Button, ButtonGroup, ButtonGroupProps, ButtonProps, Input } from '@chakra-ui/react'
+import { ReactElement, useMemo, useState } from 'react'
+import { generatePath, Link as RouterLink, useNavigate, useParams } from 'react-router-dom'
 import { usePagination, useRoutedPagination } from './PaginationProvider'
 
-export const Pagination = () => {
-  const { page, setPage, totalPages } = usePagination()
+export type PaginationProps = ButtonGroupProps & {
+  maxButtons?: number | false
+  buttonProps?: ButtonProps
+}
+
+const createButton = (page: number, currentPage: number, props: ButtonProps) => (
+  <Button key={page} isActive={currentPage === page} {...props}>
+    {page + 1}
+  </Button>
+)
+
+const createEllipsisButton = (
+  key: string,
+  onClick: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void,
+  showInput: boolean,
+  onKeyDown: (e: React.KeyboardEvent<HTMLInputElement>) => void,
+  onBlur: () => void
+) => {
+  if (showInput) {
+    return <Input key={key} placeholder='Page #' width='50px' onKeyDown={onKeyDown} onBlur={onBlur} autoFocus />
+  } else {
+    return (
+      <Button key={key} as='a' href='#goto-page' onClick={onClick}>
+        ...
+      </Button>
+    )
+  }
+}
```

Comment:
This else can be deleted

```suggestion
  if (showInput) {
    return <Input key={key} placeholder='Page #' width='50px' onKeyDown={onKeyDown} onBlur={onBlur} autoFocus />
  }
  return (
    <Button key={key} as='a' href='#goto-page' onClick={onClick}>
      ...
    </Button>
  )
}
```

Reply:
💯 Agree